frontend/src/metabase/hooks/use-debounced-function.ts

```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+
+export function useDebouncedFunction<TVALUE>(
```

Comment:
maybe `useDebouncedEffect` would be more descriptive? I was trying to stay consistent with the useDebouncedValue, but pass a function instead of a value.

see: https://www.npmjs.com/package/use-debounced-effect

Reply:
I see what you're trying to do here. I didn't realize we have https://github.com/metabase/metabase/blob/b5747678f4d3381dc803edabbebf97377e30126e/frontend/src/metabase/hooks/use-debounced-value.ts#L3

I'd just extend that hook and accept functions that produce value in addition to just accepting a value. There are a bunch of API out there that do this kind of thing, either pass a value or a function that produce value. e.g `setState` you can pass either a value or a function`. What do you think?